pkg/generate/app/cmd/resolve.go

```diff
@@ -110,14 +115,18 @@ func Resolve(r *Resolvers, c *ComponentInputs, g *GenerationInputs) (*ResolvedCo
 		repo.SetContextDir(g.ContextDir)
 	}
 
-	if len(g.Strategy) != 0 && len(repositories) == 0 && !g.BinaryBuild {
+	if g.Strategy != generate.StrategyUnspecified && len(repositories) == 0 && !g.BinaryBuild {
 		return nil, errors.New("when --strategy is specified you must provide at least one source code location")
 	}
 
 	if g.BinaryBuild && (len(repositories) > 0 || components.HasSource()) {
 		return nil, errors.New("specifying binary builds and source repositories at the same time is not allowed")
 	}
 
+	if g.BinaryBuild && g.Strategy == generate.StrategyPipeline {
+		return nil, errors.New("specifying binary builds and the pipeline strategy at the same time is not allowed")
+	}
```

Comment:
can this check not be done in the cmd validation logic?  (newapp.CompleteAppConfig() ).  That seems like the more appropriate location.